src/Core/Utility.js

```diff
@@ -0,0 +1,176 @@
+/**
+ * @constructor
+ */
+function List () {
+	this.next = this
+	this.prev = this
+	this.length = 0
+}
+List.prototype = {
```

Comment:
when overriding `prototype`, best practice is to maintain the `constructor` (I think for logging)
e.g.

```js
function List(){}
List.prototype = {
  constructor: List
  a: function(){}
}
```

Reply:
Nevermind - https://stackoverflow.com/questions/4012998/what-it-the-significance-of-the-javascript-constructor-property